src/base_classes/sports.py

```diff
@@ -28,6 +33,15 @@ def __init__(self, config: Dict[str, Any], display_manager: DisplayManager, cach
         self.display_height = self.display_manager.matrix.height
 
         self.sport_key = sport_key
+        
+        # Initialize new architecture components
+        self.sport_config = get_sport_config(sport_key, logger)
+        self.api_extractor = get_extractor_for_sport(sport_key, logger)
```

Comment:
you can do it like this or in the `Football` class just do `self.api_extractor = ESPNFootballExtractor()` and then u don't need the dictionary. Thats the nice thing of using class inheritance.